modules/renderer/features.js

```diff
@@ -529,17 +534,37 @@ export function rendererFeatures(context) {
 
 
     features.isHidden = function(entity, resolver, geometry) {
-        if (!_hidden.length) return false;
         if (!entity.version) return false;
-
+        if (!features.featureFitsDateRange(entity)) return true;
+        if (!_hidden.length) return false;
         var fn = (geometry === 'vertex' ? features.isHiddenChild : features.isHiddenFeature);
         return fn(entity, resolver, geometry);
     };
 
 
-    features.filter = function(d, resolver) {
-        if (!_hidden.length) return d;
+    features.featureFitsDateRange = function (entity) {
```

Comment:
Previously this method was implemented as a rule, which had a nice side effect that the blue chip in the status bar would include the number of features hidden for being out of range. There’s a lot of code around deciding when to recalculate the count. Patching up `gatherStats()` might do the trick.